doc/languages-frameworks/qt.section.md

```diff
@@ -0,0 +1,113 @@
+
+Qt
+=
+This section describes the differences between Nix expressions for Qt libraries and applications and Nix expressions for other C++ software. Some knowledge of the latter is assumed.
+
+There are primarily two problems which the Qt infrastructure is designed to address: ensuring consistent versioning of all dependencies and finding dependencies at runtime.
+
+**Nix expression for a Qt package (default.nix)**
+
+```nix
+{ mkDerivation, lib, qtbase }: #1
+mkDerivation { #2
```

Comment:
Hmm, we will need to figure out the callouts still. Unfortunately, the RFC was more of a declaration so we have to come up with a solution ourselves.

Reply:
Yeah it is a tricky one. Not sure what we should use instead of them.

Reply:
In https://github.com/NixOS/nixpkgs/pull/105036 I introduce Sphinx + MyST as CommonMark implementation. That way, we can fall back to rST and have at least a bit more possibilities. Unfortunately, rST does not support callouts, however see https://stackoverflow.com/questions/997428/how-to-write-callouts-using-restructured-text for reStructuredText.

Reply:
Are we able to stick with the docbook for the callouts for now?

Reply:
Having a bunch of trouble with the docbook tags. pandoc keeps stripping out the outer tags and causes syntax errors with docbook.

I am going to keep looking into it but if anyone has an example of being able to put docbook tags inline I can go off that would be much appreciated.

eg. if I put the following in to the document:

```
### Example buildBowerComponents {#ex-buildBowerComponents}

<programlisting language="nix">
bowerComponents = buildBowerComponents {
  name = "my-web-app";
  generated = ./bower-packages.nix; <co xml:id="ex-buildBowerComponents-1" />
  src = myWebApp; <co xml:id="ex-buildBowerComponents-2" />
};
</programlisting>
```

pandoc generates

```
    <section xml:id="ex-buildBowerComponents">
      <title>Example buildBowerComponents</title>
      bowerComponents = buildBowerComponents { name =
      <quote>my-web-app</quote>; generated = ./bower-packages.nix;
      <co xml:id="ex-buildBowerComponents-1" /> src = myWebApp;
      <co xml:id="ex-buildBowerComponents-2" /> };
      <para>
        In
        <link linkend="ex-buildBowerComponents"><quote>buildBowerComponents</quote>
        example</link> the following arguments are of special
        significance to the function:
      </para>

```

I have tried a bunch of different variations including a whole section but it keeps stripping tags.

Reply:
Hmm, I guess that will only work for inline tags :crying_cat_face:

Reply:
Awesome I figured it out.

If you want to put in line docbook stuff and it is stripping it like above you can do the following to stop it stripping the tags:
````
```{=docbook}
docbook markdown here
```
````

Reply:
Added the docbook callouts back in again.